core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/transport/AbstractTransportListener.java

```diff
@@ -183,7 +195,22 @@ protected EndpointReference genEpr(String protocol, String ip, String serviceCon
             String proxyContextPath = CarbonUtils.getProxyContextPath(true);
 
             String tmp = protocol + "://" + ip;
-            if (proxyPort == 80 || proxyPort == 443) {
+
+            String workerProxyPort = null;
+            if (protocol == ServerConstants.HTTP_TRANSPORT) {
```

Comment:
Shouldn't we use .equals() here ?
like,
ServerConstants.HTTP_TRANSPORT.equals(protocol)